### PR TITLE
chore: use vitest

### PR DESCRIPTION
**Pull request by lukashass**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -38,12 +38,6 @@ jobs:
       - name: Unit test
         run: pnpm run test
 
-      - uses: artiomtr/jest-coverage-report-action@v2.0-rc.1
```
Comment (anbraten) [inline .github/workflows/tests.yml:41]: Is there some alternative to that action?

File: package.json [hunk 2]
```diff
@@ -37,28 +37,25 @@
...
     "prettier": "2.6.1",
     "semantic-release": "19.0.2",
     "supertest": "6.2.2",
-    "ts-jest": "27.1.4",
-    "ts-node": "10.7.0",
     "tsup": "5.12.1",
     "typescript": "4.6.3",
-    "vite": "2.8.6"
+    "vite": "2.9.8",
+    "vitest": "0.10.5"
```
Comment (anbraten) [inline package.json:58]: Could you add https://www.npmjs.com/package/@vitest/ui.

**Review by anbraten — APPROVED** [review]

**Comment by anbraten** [comment]
@lukashass Could you move the test files from `tests` next to the source files.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (lukashass) [reply .github/workflows/tests.yml:41]: @anbraten Not really, the next best thing is the setting to enforce 100% coverage that I enabled.

File: package.json (hunk 2 above)
Comment (lukashass) [reply package.json:58]: @anbraten I would prefer not to add this to the project. After all there is https://vitest.dev/guide/ide.html

**Review by anbraten — APPROVED** [review]

**Pull request merged**